## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 490797`
Commit: [`e25df4a6ff10186c832682d0ce3678455db0562d`](https://github.com/NixOS/nixpkgs/commit/e25df4a6ff10186c832682d0ce3678455db0562d) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/e25df4a6ff10186c832682d0ce3678455db0562d..pull/490797/head))
Merge: [`2f3ffb37dca4b7ebbddd507fec83cc4b8548c83e`](https://github.com/NixOS/nixpkgs/commit/2f3ffb37dca4b7ebbddd507fec83cc4b8548c83e)

Logs: https://github.com/robertjakub/nixpkgs-review-gha/actions/runs/22118146388


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.aiotools</li>
    <li>python313Packages.aiotools.dist</li>
    <li>python314Packages.aiotools</li>
    <li>python314Packages.aiotools.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.aiotools</li>
    <li>python313Packages.aiotools.dist</li>
    <li>python314Packages.aiotools</li>
    <li>python314Packages.aiotools.dist</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python313Packages.aiotools</li>
    <li>python313Packages.aiotools.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python314Packages.aiotools</li>
    <li>python314Packages.aiotools.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python313Packages.aiotools</summary>
<pre>tests/test_ptaskgroup.py::test_ptaskgroup_current
tests/test_ptaskgroup.py::test_ptaskgroup_current
tests/test_ptaskgroup.py::test_ptaskgroup_current
  /nix/build/nix-6196-1989224250/source/tests/test_ptaskgroup.py:481: DeprecationWarning: Use aiotools.TaskScope instead.
    async with aiotools.PersistentTaskGroup(name=&quot;inner&quot;) as tg:

tests/test_ptaskgroup.py::test_ptaskgroup_enumeration
  /nix/build/nix-6196-1989224250/source/tests/test_ptaskgroup.py:507: DeprecationWarning: Use aiotools.TaskScope instead.
    async with aiotools.PersistentTaskGroup() as tg:

tests/test_ptaskgroup.py::test_ptaskgroup_enumeration
tests/test_ptaskgroup.py::test_ptaskgroup_enumeration
tests/test_ptaskgroup.py::test_ptaskgroup_enumeration
  /nix/build/nix-6196-1989224250/source/tests/test_ptaskgroup.py:501: DeprecationWarning: Use aiotools.TaskScope instead.
    async with aiotools.PersistentTaskGroup() as tg:

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_fork.py::test_fork_many[spawn-posix] - assert 100 == 101
= 1 failed, 169 passed, 20 skipped, 1 xfailed, 26 warnings in 118.44s (0:01:58) =</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.aiotools</li>
    <li>python313Packages.aiotools.dist</li>
    <li>python314Packages.aiotools</li>
    <li>python314Packages.aiotools.dist</li>
  </ul>
</details>

--- a/pkgs/development/python-modules/aiotools/default.nix
+++ b/pkgs/development/python-modules/aiotools/default.nix
@@ -1,0 +1,45 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools-scm,
+  async-lru,
+  pytestCheckHook,
+  pytest-asyncio,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "aiotools";
+  version = "2.2.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "achimnol";
+    repo = "aiotools";
+    tag = finalAttrs.version;
+    hash = "sha256-uIG3JPqep4NGtZa7Qo8SOK9Ca1GNKyuBasFtwR9oG8U=";
+  };
+
+  build-system = [
+    setuptools-scm
+  ];
+
+  dependencies = [
+    async-lru
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-asyncio
+  ];
+
+  pythonImportsCheck = [ "aiotools" ];
+
+  meta = {
+    description = "Idiomatic asyncio utilities";
+    homepage = "https://github.com/achimnol/aiotools";
+    changelog = "https://github.com/achimnol/aiotools/blob/${finalAttrs.src.tag}/CHANGES.md";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ robertjakub ];
+  };
+})

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -536,6 +536,8 @@ self: super: with self; {
 
   aiotedee = callPackage ../development/python-modules/aiotedee { };
 
+  aiotools = callPackage ../development/python-modules/aiotools { };
+
   aiotractive = callPackage ../development/python-modules/aiotractive { };
 
   aiounifi = callPackage ../development/python-modules/aiounifi { };

